src/signalrclient/connection_impl.cpp

```diff
@@ -345,6 +368,24 @@ namespace signalr
                     // we do let the exception through (especially the task_canceled exception)
                     connection->m_transport = nullptr;
                 }
+
+                try
+                {
+                    connection->m_disconnected();

```

Comment:
This is now called when we hold the lock - we should release the lock before calling user's code.